spec/unit/rom/reader_spec.rb

```diff
@@ -10,6 +10,55 @@
   let(:mappers) { ROM::MapperRegistry.new(users: mapper) }
   let(:mapper) { double('mapper', header: []) }
 
+  describe '.build' do
+    subject(:reader) do
+      ROM::Reader.build(name, relation, mappers, [:all])
+    end
+
+    before do
+      relation.instance_exec do
+        def name
+          'users'
+        end
+
+        def all(*_args)
+          find_all
+        end
+      end
+    end
+
+    it 'sets reader class name' do
+      expect(reader.class.name).to eql("ROM::Reader[Users]")
+    end
+
+    it 'defines methods from relation' do
+      block = proc {}
+
+      expect(relation).to receive(:all)
+        .with(1, &block)
+        .and_return([joe])
+
+      expect(mapper).to receive(:process)
+        .with([joe])
+        .and_yield(joe)
+
+      result = reader.all(1, &block)
```

Comment:
There appears to be a relationship between the the integer `1` here and on line `38`, but would it be clearer to extract it to an explaining variable within this example?